.bazelrc

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# DISCLAIMER: Bazel support is community-based. The maintainers do not
```

Comment:
Is there any way to gather all (or part of) Bazel files in a single subdirectory? Are there any best practices or conventions regarding this?

It may be useful for files organization and for README.md inside this folder, just like https://github.com/oneapi-src/oneTBB/tree/master/cmake#readme
It would be a good place for getting started info and for this disclaimer.

Reply:
The `WORKSPACE.bazel` file needs to life at the root (details [here](https://docs.bazel.build/versions/main/build-ref.html#workspace)). Otherwise, the above shown [demo](https://github.com/Vertexwahn/BazelDemos/tree/master/oneTBB) will not work - other Bazel-based projects will be not able to easily fetch oneTBB as a dependency. Every Bazel-based project has always a WORKSPACE file at its root.
 Also, the `BUILD.bazel` file cannot be moved - since we need to reference source files at different places.
Putting all the "Bazel stuff" in one directory does not work.
`.bazelversion` and `.bazelrc` can also not be moved but they a kind of optional - not mandatory. `.bazelversion` just holds the information which Bazel version has been used/aimed for. Bazelisk (a frequently used tool within Bazel community)  picks up the defined Bazel version in `.bazelversion` and uses it to build. Currently, the Bazelization works also with earlier versions of Bazel (e.g. 3.5.0). Personally, I like to document what Bazel version was used - but if you want to minimize root files we could get rid of it. `bazel.rc` allows to set general options for Bazel - I defined an out of source build here. But this is not of importance for a library user. Nevertheless, I think both files do not hurt and propose to keep them. But if you wish I can remove them. All further files (additional scripts for some magic) can go to a dedicated `bazel` folder (e.g. fetching other dependencies, helper scripts, etc.) - similar to the `cmake` folder.

Reply:
I will add a `bazel `folder with an REAMDE.md

Reply:
You explained quite well, thank you! I'm not insisting on minimizing Bazel files in the root just for fun. If there is a reason to have such files then I'm ok, I was curious in terms of layout organization. If all the Bazel files should be in root, maybe it doesn't make sense to create `bazel` folder just for README. Could you write some get started / intro for the Bazel in oneTBB? Depending on the size we can consider different options where to place it:
- In root README.md if it is short enough
- In some bazel.md with reference from root README.md
- In bazel/README.md with reference from root README.md
- Any other ideas?

Reply:
I added a `Bazel.md` in the root folder